src/main/java/de/tum/in/www1/artemis/service/metis/PostService.java

```diff
@@ -557,6 +583,16 @@ private void mayInteractWithPostElseThrow(Post post, User user, Course course) {
         }
     }
 
+    private void mayInteractWithConversationElseThrow(Long conversationId) {
+        final User user = userRepository.getUserWithGroupsAndAuthorities();
+        Conversation conversation = conversationService.getConversationById(conversationId);
+
+        if (conversation == null
+                || conversation.getConversationParticipants().stream().noneMatch(conversationParticipant -> conversationParticipant.getUser().getId().equals(user.getId()))) {
+            throw new BadRequestAlertException("User not authorized to access this conversation!", METIS_POST_ENTITY_NAME, "");
```

Comment:
This is not really a Bad Request, more an Unauthorized Request, right?

Reply:
It's actually kind of misuse. Unauthorized means that the user is not allowed because of missing required authorization. Forbidden is usually unallowed independent of the authorization, e.g. only for internal use.

You could implement a UnauthorizedRequestException analog to the AccessForbiddenException (actually not a lot of work) but it would also be fine if you use the AccessForbiddenException.

Reply:
fixed